airflow/models/xcom.py

```diff
@@ -99,29 +137,63 @@ def set(cls, key, value, task_id, dag_id, execution_date=None, run_id=None, sess
 
         value = XCom.serialize_value(value)
 
-        # remove any duplicate XComs
+        # Remove duplicate XComs and insert a new one.
         session.query(cls).filter(
-            cls.key == key, cls.execution_date == execution_date, cls.task_id == task_id, cls.dag_id == dag_id
+            cls.key == key,
+            cls.execution_date == execution_date,
+            cls.task_id == task_id,
+            cls.dag_id == dag_id,
         ).delete()
-
+        new = cast(Any, cls)(  # Work around Mypy complaining model not defining '__init__'.
+            key=key,
+            value=cls.serialize_value(value),
```

Comment:
This is different -- before it was just doing `value=value`.

```suggestion
            value=value,
```

I think?

Reply:
Hmm not sure how this got here.

Reply:
Ah I see, wrong copy-paste. I merged line 100/138 above to this in the other PR, but forgot to delete that line when copying the new implementation.